samples/flogs/LogsTask.cpp

```diff
@@ -75,17 +77,26 @@ bool LogsTask::WriteLogs() {
     DARWIN_LOGGER;
     DARWIN_LOG_DEBUG("WriteLogsTask::Write:: Starting writing in log file: \""
                      + _log_file_path + "\"...");
+    unsigned int retry = RETRY;
+    bool fail;
+
+    fail = !_log_file.is_open() or _log_file.fail();
+    while(retry and fail){
+        DARWIN_LOG_INFO("LogsGenerator::LoadClassifier:: Error when opening the log file, "
+                        "will retry " + std::to_string(retry) + " times");
+        _log_file.open(_log_file_path, std::ios::out | std::ios::app);
+        fail = !_log_file.is_open() or _log_file.fail();
+        retry--;
+    }
 
-    std::ofstream logFile(_log_file_path, std::ios::out | std::ios::app);
-
-    if (!logFile.is_open() or logFile.fail()) {
-        DARWIN_LOG_ERROR("WriteLogsTask::Write:: Error when opening the log file, "
-                         "maybe too low space disk or wrong permission");
+    if(fail) {
+        DARWIN_LOG_ERROR("LogsGenerator::LoadClassifier:: Error when opening the log file, "
+                         "too many retry, "
+                         "may due to low space disk or wrong permission");
         return false;
     }
 
-    logFile << body;
-    logFile.close();
+    _log_file << body << std::flush;
 
```

Comment:
logFile.close() missing ?

Reply:
No, it is part of the improvement.
Previously we closed the file after each input and opened it again to write the next one, now we keep the stream open until end or failure.